### ISSUE TITLE
Queue.deserialize invocation error

### ISSUE THREAD

**Issue by andrewvc** [elastic/logstash#9104]
This has been spotted in the wild here: https://github.com/logstash-plugins/logstash-codec-multiline/issues/58#issuecomment-361667313 where it was incorrectly labeled AFAICT.

It looks like a bug in Queue.deserialize .  Any ideas @colinsurprenant @guyboertje @danhermann as to what's triggering this?

```
[2018-01-30T07:23:02,624][DEBUG][logstash.util.decorators ] inputs/LogStash::Inputs::Udp: adding value to field {"field"=>"[@metadata][inputtype]", "value"=>["truxetrace"]}

Exception in thread "[main]>worker0" org.logstash.ackedqueue.QueueRuntimeException: deserialize invocation error

at org.logstash.ackedqueue.Queue.deserialize(Queue.java:638)

at org.logstash.ackedqueue.Page.readBatch(Page.java:60)

at org.logstash.ackedqueue.Queue._readPageBatch(Queue.java:505)

at org.logstash.ackedqueue.Queue.readBatch(Queue.java:496)

at org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue.ruby_read_batch(JrubyAckedQueueExtLibrary.java:162)

at org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$2$0$ruby_read_batch.call(JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$2$0$ruby_read_batch.gen)

at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:202)

at rubyjit.LogStash::Util::WrappedAckedQueue$$read_batch_d49228611d045c1563109c40e125e11f223ebb3d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:86)

at rubyjit.LogStash::Util::WrappedAckedQueue$$read_batch_d49228611d045c1563109c40e125e11f223ebb3d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)

at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:221)

at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:202)

at rubyjit.LogStash::Util::WrappedAckedQueue::ReadBatch$$read_next_b8fddaedce1c7efd76622fff6ba60df85ed33b8d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:275)

at rubyjit.LogStash::Util::WrappedAckedQueue::ReadBatch$$read_next_b8fddaedce1c7efd76622fff6ba60df85ed33b8d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)

at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:141)

at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)

at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.chained_0_ensure_1$RUBY$__ensure__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:186)

at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:185)

at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)

...

 

ERROR #2

[2018-01-30T07:39:02,728][DEBUG][logstash.outputs.file  ] 0 stale files found {:inactive_files=>{}}

Exception in thread "[main]>worker1" org.logstash.ackedqueue.QueueRuntimeException: deserialize invocation error

    at org.logstash.ackedqueue.Queue.deserialize(Queue.java:638)

    at org.logstash.ackedqueue.Page.readBatch(Page.java:60)

    at org.logstash.ackedqueue.Queue._readPageBatch(Queue.java:505)

    at org.logstash.ackedqueue.Queue.readBatch(Queue.java:496)

    at org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue.ruby_read_batch(JrubyAckedQueueExtLibrary.java:162)
```

**Comment by colinsurprenant** [comment]
It would be beneficial to have the exact environment information from @jkelastic like LS version and OS.

**Comment by colinsurprenant** [comment]
So this could be related to #8379 - this fix was backported down to 5.5, in 5.5.4. It seems @jkelastic is running 5.5 but I haven't found the exact bug release version. Lets first get an acknowledgement on this before further investigation.

**Comment by colinsurprenant** [comment]
Ok, found the complete stacktrace in the logs referenced in logstash-plugins/logstash-codec-multiline/issues/58

```
Exception in thread "[main]>worker0" org.logstash.ackedqueue.QueueRuntimeException: deserialize invocation error
	at org.logstash.ackedqueue.Queue.deserialize(Queue.java:638)
	at org.logstash.ackedqueue.Page.readBatch(Page.java:60)
	at org.logstash.ackedqueue.Queue._readPageBatch(Queue.java:505)
	at org.logstash.ackedqueue.Queue.readBatch(Queue.java:496)
	at org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue.ruby_read_batch(JrubyAckedQueueExtLibrary.java:162)
	at org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$2$0$ruby_read_batch.call(JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$2$0$ruby_read_batch.gen)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:202)
	at rubyjit.LogStash::Util::WrappedAckedQueue$$read_batch_d49228611d045c1563109c40e125e11f223ebb3d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:86)
	at rubyjit.LogStash::Util::WrappedAckedQueue$$read_batch_d49228611d045c1563109c40e125e11f223ebb3d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)
	at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:221)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:202)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadBatch$$read_next_b8fddaedce1c7efd76622fff6ba60df85ed33b8d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:275)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadBatch$$read_next_b8fddaedce1c7efd76622fff6ba60df85ed33b8d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)
	at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:141)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.chained_0_ensure_1$RUBY$__ensure__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:186)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:185)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)
	at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:141)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)
	at org.jruby.ast.CallNoArgNode.interpret(CallNoArgNode.java:60)
	at org.jruby.ast.LocalAsgnNode.interpret(LocalAsgnNode.java:123)
	at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
	at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
	at org.jruby.ast.WhileNode.interpret(WhileNode.java:131)
	at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
	at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
	at org.jruby.evaluator.ASTInterpreter.INTERPRET_METHOD(ASTInterpreter.java:74)
	at org.jruby.internal.runtime.methods.InterpretedMethod.call(InterpretedMethod.java:225)
	at org.jruby.internal.runtime.methods.DefaultMethod.call(DefaultMethod.java:219)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:346)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:204)
	at org.jruby.ast.FCallTwoArgNode.interpret(FCallTwoArgNode.java:38)
	at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
	at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
	at org.jruby.evaluator.ASTInterpreter.INTERPRET_BLOCK(ASTInterpreter.java:112)
	at org.jruby.runtime.Interpreted19Block.evalBlockBody(Interpreted19Block.java:206)
	at org.jruby.runtime.Interpreted19Block.yield(Interpreted19Block.java:194)
	at org.jruby.runtime.Interpreted19Block.call(Interpreted19Block.java:125)
	at org.jruby.runtime.Block.call(Block.java:101)
	at org.jruby.RubyProc.call(RubyProc.java:300)
	at org.jruby.RubyProc.call(RubyProc.java:230)
	at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:103)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor25.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.logstash.ackedqueue.Queue.deserialize(Queue.java:636)
	... 43 more
Caused by: java.lang.IllegalArgumentException: Missing Valuefier handling for full class name=[B, simple name=byte[]
	at org.logstash.Valuefier.convertNonCollection(Valuefier.java:51)
	at org.logstash.Valuefier.convert(Valuefier.java:90)
	at org.logstash.ConvertedMap.newFromMap(ConvertedMap.java:31)
	at org.logstash.Valuefier.convert(Valuefier.java:69)
	at org.logstash.Event.<init>(Event.java:62)
	at org.logstash.Event.fromSerializableMap(Event.java:208)
	at org.logstash.Event.deserialize(Event.java:395)
	... 47 more
Caused by: java.lang.IllegalArgumentException: No enum constant org.logstash.bivalues.BiValues.[B
	at java.lang.Enum.valueOf(Enum.java:238)
	at org.logstash.bivalues.BiValues.valueOf(BiValues.java:20)
	at org.logstash.bivalues.BiValues.newBiValue(BiValues.java:88)
	at org.logstash.Valuefier.convertNonCollection(Valuefier.java:48)
	... 53 more
Exception in thread "[main]>worker1" org.logstash.ackedqueue.QueueRuntimeException: deserialize invocation error
	at org.logstash.ackedqueue.Queue.deserialize(Queue.java:638)
	at org.logstash.ackedqueue.Page.readBatch(Page.java:60)
	at org.logstash.ackedqueue.Queue._readPageBatch(Queue.java:505)
	at org.logstash.ackedqueue.Queue.readBatch(Queue.java:496)
	at org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue.ruby_read_batch(JrubyAckedQueueExtLibrary.java:162)
	at org.logstash.ackedqueue.ext.JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$2$0$ruby_read_batch.call(JrubyAckedQueueExtLibrary$RubyAckedQueue$INVOKER$i$2$0$ruby_read_batch.gen)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:202)
	at rubyjit.LogStash::Util::WrappedAckedQueue$$read_batch_d49228611d045c1563109c40e125e11f223ebb3d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:86)
	at rubyjit.LogStash::Util::WrappedAckedQueue$$read_batch_d49228611d045c1563109c40e125e11f223ebb3d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)
	at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:221)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:202)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadBatch$$read_next_b8fddaedce1c7efd76622fff6ba60df85ed33b8d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:275)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadBatch$$read_next_b8fddaedce1c7efd76622fff6ba60df85ed33b8d1808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)
	at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:141)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.chained_0_ensure_1$RUBY$__ensure__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:186)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb:185)
	at rubyjit.LogStash::Util::WrappedAckedQueue::ReadClient$$read_batch_fcf4821efe51d29d2ff1b09c121b6d1cbbd684311808253012.__file__(E:/Logstash/logstash-core/lib/logstash/util/wrapped_acked_queue.rb)
	at org.jruby.internal.runtime.methods.JittedMethod.call(JittedMethod.java:141)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:134)
	at org.jruby.ast.CallNoArgNode.interpret(CallNoArgNode.java:60)
	at org.jruby.ast.LocalAsgnNode.interpret(LocalAsgnNode.java:123)
	at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
	at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
	at org.jruby.ast.WhileNode.interpret(WhileNode.java:131)
	at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
	at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
	at org.jruby.evaluator.ASTInterpreter.INTERPRET_METHOD(ASTInterpreter.java:74)
	at org.jruby.internal.runtime.methods.InterpretedMethod.call(InterpretedMethod.java:225)
	at org.jruby.internal.runtime.methods.DefaultMethod.call(DefaultMethod.java:219)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:202)
	at org.jruby.ast.FCallTwoArgNode.interpret(FCallTwoArgNode.java:38)
	at org.jruby.ast.NewlineNode.interpret(NewlineNode.java:105)
	at org.jruby.ast.BlockNode.interpret(BlockNode.java:71)
	at org.jruby.evaluator.ASTInterpreter.INTERPRET_BLOCK(ASTInterpreter.java:112)
	at org.jruby.runtime.Interpreted19Block.evalBlockBody(Interpreted19Block.java:206)
	at org.jruby.runtime.Interpreted19Block.yield(Interpreted19Block.java:194)
	at org.jruby.runtime.Interpreted19Block.call(Interpreted19Block.java:125)
	at org.jruby.runtime.Block.call(Block.java:101)
	at org.jruby.RubyProc.call(RubyProc.java:300)
	at org.jruby.RubyProc.call(RubyProc.java:230)
	at org.jruby.internal.runtime.RubyRunnable.run(RubyRunnable.java:103)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.GeneratedMethodAccessor25.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.logstash.ackedqueue.Queue.deserialize(Queue.java:636)
	... 42 more
Caused by: java.lang.IllegalArgumentException: Missing Valuefier handling for full class name=[B, simple name=byte[]
	at org.logstash.Valuefier.convertNonCollection(Valuefier.java:51)
	at org.logstash.Valuefier.convert(Valuefier.java:90)
	at org.logstash.ConvertedMap.newFromMap(ConvertedMap.java:31)
	at org.logstash.Valuefier.convert(Valuefier.java:69)
	at org.logstash.Event.<init>(Event.java:62)
	at org.logstash.Event.fromSerializableMap(Event.java:208)
	at org.logstash.Event.deserialize(Event.java:395)
	... 46 more
Caused by: java.lang.IllegalArgumentException: No enum constant org.logstash.bivalues.BiValues.[B
	at java.lang.Enum.valueOf(Enum.java:238)
	at org.logstash.bivalues.BiValues.valueOf(BiValues.java:20)
	at org.logstash.bivalues.BiValues.newBiValue(BiValues.java:88)
	at org.logstash.Valuefier.convertNonCollection(Valuefier.java:48)
```

Where the interesting bit are:
```
Caused by: java.lang.IllegalArgumentException: Missing Valuefier handling for full class name=[B, simple name=byte[]
	at org.logstash.Valuefier.convertNonCollection(Valuefier.java:51)
	at org.logstash.Valuefier.convert(Valuefier.java:90)
	at org.logstash.ConvertedMap.newFromMap(ConvertedMap.java:31)
	at org.logstash.Valuefier.convert(Valuefier.java:69)
	at org.logstash.Event.<init>(Event.java:62)
	at org.logstash.Event.fromSerializableMap(Event.java:208)
	at org.logstash.Event.deserialize(Event.java:395)
	... 47 more
Caused by: java.lang.IllegalArgumentException: No enum constant org.logstash.bivalues.BiValues.[B
	at java.lang.Enum.valueOf(Enum.java:238)
	at org.logstash.bivalues.BiValues.valueOf(BiValues.java:20)
	at org.logstash.bivalues.BiValues.newBiValue(BiValues.java:88)
	at org.logstash.Valuefier.convertNonCollection(Valuefier.java:48)
	... 53 more
```

which definitely point to #8379 I mentioned above. 

I also see in the debug trace that the event contains the following field
```
             "sqlplan" => "0x06001D0014260A35A017519B0400000001000000000000000000000000000000000000000000000000000000",
```
which also point at the same  bignum deserialization bug.

Let's wait for the version information but so far everything points at @jkelastic running LS version <  5.5.4 and the fix will be to upgrade to the latest 5.6.7 or 6.2.0.

**Comment by andrewvc** [comment]
Should be fixed as mentioned in the last comment.
